src/devtools/views/Profiler/utils.js

```diff
@@ -28,8 +28,11 @@ export const calculateSelfDuration = (
   }
 
   let selfDuration = ((actualDurations.get(id): any): number);
+  const node = nodes.get(id);
+  if (node == null) {
+    throw Error(`Could not find node with id "${id}" in commit tree`);
+  }
```

Comment:
FWIW, I didn't add explicit checks for things like this initially because the line immediately below (`node.children`) would error anyway.

Reply:
> FWIW, I didn't add explicit checks for things like this initially because the line immediately below (node.children) would error anyway.

I prefer to not leave a possibility for obscure errors like "cannot read property children of null" if I can. And with Flow/TypeScript I can.